tests/integration/test_create_token.py

```diff
@@ -0,0 +1,138 @@
+import uuid
+
+from app.authentication.auth_payload_versions import AuthPayloadVersion
+from app.authentication.authenticator import decrypt_token
+from tests.integration.app_context_test_case import AppContextTestCase
+from tests.integration.create_token import (
+    ACCOUNT_SERVICE_URL,
+    PAYLOAD_V2_SOCIAL,
+    PAYLOAD_V2_SUPPLEMENTARY_DATA,
+)
+from tests.integration.integration_test_case import IntegrationTestCase
+
+test_parameters = [
+    {
+        "schema": "test_metadata_routing.json",
+        "theme": "default",
+        "additional_payload": {
+            "flag_1": 123,
+            "period_id": "abc",
+            "link": "https://example.com",
+        },
+        "payload": {
+            "version": AuthPayloadVersion.V2.value,
+            "survey_metadata": {
+                "data": {
+                    "user_id": "integration-test",
+                    "period_str": "April 2016",
+                    "period_id": "201604",
+                    "ru_ref": "12345678901A",
+                    "ru_name": "Integration Testing",
+                    "ref_p_start_date": "2016-04-01",
+                    "ref_p_end_date": "2016-04-30",
+                    "trad_as": "Integration Tests",
+                    "employment_date": "1983-06-02",
+                    "display_address": "68 Abingdon Road, Goathill",
+                }
+            },
+            "collection_exercise_sid": "789",
+            "response_id": "1234567890123456",
+            "language_code": "en",
+            "roles": [],
+            "account_service_url": ACCOUNT_SERVICE_URL,
+        },
+    },
+    {
+        "schema": "social_demo.json",
+        "theme": "social",
+        "additional_payload": {
+            "flag_1": True,
+            "user_id": "abc",
+            "date": "2016-05-12",
+        },
+        "payload": {
+            "version": AuthPayloadVersion.V2.value,
+            "survey_metadata": {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": str(uuid.uuid4()),
+                },
+                "receipting_keys": ["qid"],
+            },
+            "collection_exercise_sid": "789",
+            "response_id": "1234567890123456",
+            "language_code": "en",
+            "roles": [],
+            "account_service_url": ACCOUNT_SERVICE_URL,
+        },
+    },
+]
+
+"""
```

Comment:
Would move this comment so it sits under the class, also minor I'd prob remove the v1/v2 stuff since there is already a ticket raised to remove it so won't be around for much longer.